sound/soc/sof/ipc4-topology.c

```diff
@@ -1129,38 +1169,12 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 		return i;
 	}
 
-	/* find pin_index of the sink widget belonging to the DAPM widget list */
-	snd_soc_dapm_widget_for_each_sink_path(swidget->widget, p) {
-		if (!widget_in_list(list, p->sink))
-			continue;
-
-		for (i = 0; i < num_output_pins; i++) {
-			if (!strcmp(output_pin_binding[i], p->sink->name)) {
-				pin_index = i;
-				break;
-			}
-		}
-
-		if (pin_index >= 0)
-			break;
-	}
-
-	if (pin_index < 0) {
-		dev_err(sdev->dev, "No matching output pin index found for widget %s\n",
-			swidget->widget->name);
-		return -EINVAL;
-	}
-
-	for (i = 0; i < available_fmt->num_output_formats; i++) {
-		if (available_fmt->output_pin_fmts[i].pin_index == pin_index) {
-			base_config->obs = available_fmt->output_pin_fmts[i].buffer_size;
-			return i;
-		}
-	}
+	/* Return format index of the output format based on output pin binding */
+	ret = sof_ipc4_get_output_format(sdev, swidget, available_fmt, list);
+	if (ret >= 0)
+		base_config->obs = available_fmt->output_pin_fmts[ret].buffer_size;
```

Comment:
I find this commit very confusing because it uses the same example as the previous patch.

I thought the previous patch would be good enough, but now for the same example we need another change, and I am not clear from the commit messages what is fixed by what patch.